### DISCUSSION TITLE
[vcpkg-tool] Cycle detected error message is unhelpful

### DISCUSSION THREAD

**Discussion by horenmar** [microsoft/vcpkg#21908] [Ideas]
**Is your feature request related to a problem? Please describe.**
Today I messed up my vcpkg.json a bit when trying to make one feature dependent on another, and got this error message:

```
Error: Cycle detected during onnxruntime:x64-windows-pexlib-custom:
pexlib:x64-windows-pexlib-custom
```

This is not helpful to debugging the issue.

**Proposed solution**
Print out packages that are part of the detected cycle. This is relatively easy graph problem, if we do not want to spam too much output to the user in case of failure, we can add a check on the size of the cycle and if the cycle size is e.g. >10, print out something like
```
Error: Dependency cycle of size XYZ detected, omitting parts of the cycle
Cycle: A -> B -> C -> ... -> foo -> bar -> quux -> A

**Comment by autoantwort** [comment]
Can you post an example file?

**Reply by horenmar** [reply]
I finally got around doing this. I built a simple self-contained port set through overlay ports, and played around with different possible cycles.

With a graph `root -> A -> B -> C -> B`, the output with current (2022-02-03-457b0309cefb58fae0dcb2e4c06de9bd7582967f) version of vcpkg-tool is
```
Error: Cycle detected during b:x64-windows:
a:x64-windows
b:x64-windows
c:x64-windows
```

so it tells me which package went wrong, but how do I know that the issue is b -> c -> b? 🤷 

[This is a toy setup to play with](https://github.com/microsoft/vcpkg/files/8035876/vcpkg-repro-cycle.zip)
